### PR TITLE
Sort BaseDeviceModel by device name by default

### LINKED ISSUE #1299: Sort device lists in alphabetical order
As reported:

"In GUI V2, when you click on items in Overview page and get a list of devices, could these be sorted alphabetically.. Examples of Solar and Batteries.. Turning these into an alphabetical list makes life easier."

Some device lists are already sorted alphabetically, but not all of them, so gui-v2 should do this more consistently, and sort all user-visible device lists alphabetically. Currently AggregateDeviceModel can be sorted alphabetically but not BaseDeviceModel, which creates this confusion.

Note the inverter/charger and generator models are special as gui-v2 needs to find the device with the lowest Device Instance in these cases. (Note, the inverter/charger list shown in the overview drilldowns is already sorted alphabetically.)

[Edit: tanks and temp sensors in the Levels page should be sorted as well.]

### PR DESCRIPTION
**Pull request by blammit**
All models that are used for displaying device lists should be sorted by device name. Only the inverter/charger and generator models need to be sorted solely by device instance, as this is used to determine the "first" or "main" inverter/charger and generator.

Fixes #1299

### CODE REVIEW

File: src/basedevicemodel.h [hunk 1]
```diff
@@ -91,14 +91,23 @@ class BaseDeviceModel : public QAbstractListModel
 		DeviceRole = Qt::UserRole
 	};
 
+	enum SortBy {
+		SortByDeviceName = 0x1,
```
Comment (DanielMcInnes) [inline src/basedevicemodel.h:95]: is 'SortByDeviceName' also in ascending order?

File: src/basedevicemodel.cpp [hunk 2]
```diff
@@ -290,18 +318,76 @@ BaseDevice *BaseDeviceModel::deviceAt(int index) const
...
+			// Move the entry to be immediately before this item.
+			toIndex = i > fromIndex ? i - 1 : i;
+			break;
+		}
+	}
+
+	if (fromIndex != toIndex) {
+		const int destIndex = toIndex > fromIndex ? toIndex + 1 : toIndex;
+		beginMoveRows(QModelIndex(), fromIndex, fromIndex, QModelIndex(), destIndex);
+		m_devices.move(fromIndex, toIndex);
```
Comment (DanielMcInnes) [inline src/basedevicemodel.cpp:352]: I am not familiar with QVector::move(). If we have 3 devices, `["charlie", "echo", "golf"]`, and "charlie" changes to "zulu", do we end up with:
`[ null, "echo", "golf", "zulu"]`
or:
`[ "echo", "golf", "zulu"]` ?

**Review by DanielMcInnes — COMMENTED** [review]
Do you think BaseDeviceModel should have unit tests?

**Review by DanielMcInnes — APPROVED** [review]

File: src/basedevicemodel.h (hunk 1 above)
Comment (blammit) [reply src/basedevicemodel.h:95]: Yes, also ascending.

**Pull request merged**